aten/src/ATen/native/NativeFunctions.cpp

```diff
@@ -584,5 +586,77 @@ Tensor RoiPooling2d_backward_cpu(
   throw std::runtime_error("not implemented");
 }
 
+
+
+// TODO Replace this with more accurate digamma().
+template <typename Scalar>
+static inline Scalar digamma_one(Scalar x) {
```

Comment:
Can we move the gamma stuff to its own file? NativeFunctions is getting unwieldy. It's already over 500 lines of functions that are not very related to each other.

Reply:
Yes, my current plan is to get this in, then the GPU version, then clean up the CPU dispatch macros, then separate out the native functions to their rightful places.  Let me know if you object.